### PR TITLE
DataFlash: factor out a Log_Write_Baro_instance

### PR DESCRIPTION
**Pull request by peterbarker**

### CODE REVIEW

File: libraries/DataFlash/LogFile.cpp [hunk 1]
```diff
@@ -366,8 +366,7 @@ void DataFlash_Class::Log_Write_RSSI(AP_RSSI &rssi)
     WriteBlock(&pkt, sizeof(pkt));
 }
 
-// Write a BARO packet
-void DataFlash_Class::Log_Write_Baro(AP_Baro &baro, uint64_t time_us)
+void DataFlash_Class::Log_Write_Baro_instance(AP_Baro &baro, uint64_t time_us, uint8_t baro_instance, enum LogMessages type)
 {
     if (time_us == 0) {
         time_us = AP_HAL::micros64();
```
Comment (OXINARF) [inline libraries/DataFlash/LogFile.cpp]: Keeping this in the *instance* version of the method means that if `time_us` isn't provided to the *generic* method, then the baro messages will have different time. I don't have a strong preference, but I think that keeping all messages with the same time is better, so I suggest this piece of code should be in the *generic* method.

File: libraries/DataFlash/DataFlash.h [hunk 2]
```diff
@@ -273,6 +273,8 @@ class DataFlash_Class
     void Log_Write_EKF3(AP_AHRS_NavEKF &ahrs);
 #endif
 
+    void Log_Write_Baro_instance(AP_Baro &baro, uint64_t time_us, uint8_t baro_instance, enum LogMessages type);
+
     void backend_starting_new_log(const DataFlash_Backend *backend);
 
 private:
```
Comment (OXINARF) [inline libraries/DataFlash/DataFlash.h]: Unrelated to this PR, but why do we have `private:` twice? It confused me, I first thought the new method was being declared as public.

**Comment by peterbarker** [comment]
On Thu, 28 Sep 2017, Francisco Ferreira wrote:

> Keeping this in the instance version of the method means that if time_us
> isn't provided to the generic method, then the baro messages will have
> different time. I don't have a strong preference, but I think that keeping
> all messages with the same time is better, so I suggest this piece of code
> should be in the generic method.

Changed.

> Unrelated to this PR, but why do we have private: twice? It confused me, I
> first thought the new method was being declared as public.

Redundancy?  In case the first one fails?

**Comment by OXINARF** [comment]
> Redundancy?  In case the first one fails?

😄

Merged, thanks!

**Pull request merged**